hack/collapse/auto_collapse.sh

```diff
@@ -0,0 +1,112 @@
+#!/bin/bash
+
+# Auto Collapse
+#This is a script that takes in
+# - a markdown file
+# - a designated output file
+# - the number of lines as a threshold on when to collapse a section.
+#
+# Sample
+# ./auto_collapse.sh test_example.md test_example_collapsible.md 10
+
+# Input and output files
+INPUT_FILE=${1:-"README.md"}
+OUTPUT_FILE=${2:-"README_collapsible.md"}
+THRESHOLD=${3:-10}
+
+# Ensure output file is empty initially
+rm -f "$OUTPUT_FILE"
+
+# Variables to track sections
+inside_section=false
+section_lines=()
+section_header=""
+
+INPUT="$(cat "$INPUT_FILE")"
+FULL_CHANGELOG=$(grep -E "Full Changelog" < "$INPUT_FILE")
+
+if [[ -z $FULL_CHANGELOG ]]; then
+    echo "Full Changelog not found in the input file."
+else
+    echo "Full Changelog found in the input file."
+    INPUT=$(echo "$INPUT" | sed '/Full Changelog/d')
+fi
+
+# Function to count changes (lines starting with '*')
+count_changes() {
```

Comment:
`grep "^*" $INPUT_FILE | wc -l` does the same job and looks a bit easier (and we avoid array expansion in bash :D). What do you think?

Reply:
If it is on a variable and not a file, you can use `grep "^*" <<< "${lines}" | wc -l`

Reply:
Can you try this `num_changes="$(IFS=$'\n'; grep "^*" <<< "${lines[*]}" | wc -l)"`